### PR TITLE
Boot Error Page to help with common config issues

### PR DESCRIPTION
**Pull request by jasoncalabrese**
screenshot 2016-04-23 17 55 12

screenshot 2016-04-23 18 01 57

screenshot 2016-04-23 18 01 23

### CODE REVIEW

File: lib/booterror.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+var _ = require('lodash');
+
+var head = '<!DOCTYPE html><html><head><title>Nightscout - Boot Error</title></head><body><h1>Nightscout - Boot Error</h1><dl>';
+var tail = '</dl></body></html>';
+
+function bootError(ctx) {
+
+  return function pageHandler (req, res) {
+    var errors = _.map(ctx.bootErrors, function (obj) {
```
Comment (jasoncalabrese) [inline lib/booterror.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2453657593/issues/source?bid=2452379&fileBranchId=3249620#l11)

File: lib/booterror.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+var _ = require('lodash');
+
+var head = '<!DOCTYPE html><html><head><title>Nightscout - Boot Error</title></head><body><h1>Nightscout - Boot Error</h1><dl>';
+var tail = '</dl></body></html>';
+
+function bootError(ctx) {
```
Comment (jasoncalabrese) [inline lib/booterror.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2453657593/issues/source?bid=2452379&fileBranchId=3249620#l8)

File: lib/booterror.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  return function pageHandler (req, res) {
+    var errors = _.map(ctx.bootErrors, function (obj) {
+      obj.err = _.pick(obj.err, Object.getOwnPropertyNames(obj.err));
+      return '<dt>' + obj.desc + '</dt><dd>' + JSON.stringify(obj.err).replace(/\\n/g, '<br/>') + '</dd>';
+    }).join(' ');
+
+    res.set('Content-Type', 'text/html');
+    res.send(head + errors + tail);
+
+  }
```
Comment (jasoncalabrese) [inline lib/booterror.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2453657593/issues/source?bid=2452379&fileBranchId=3249620#l19)

File: lib/booterror.js (hunk 1 above)
Comment (jasoncalabrese) [inline lib/booterror.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2453684649/issues/source?bid=2452379&fileBranchId=3249620#l11)

File: lib/booterror.js (hunk 2 above)
Comment (jasoncalabrese) [inline lib/booterror.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2453684649/issues/source?bid=2452379&fileBranchId=3249620#l8)

File: lib/booterror.js (hunk 3 above)
Comment (jasoncalabrese) [inline lib/booterror.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2453684649/issues/source?bid=2452379&fileBranchId=3249620#l19)

Comment (jasoncalabrese) [inline lib/booterror.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2453712997/issues/source?bid=2452379&fileBranchId=3249620#l19)

File: lib/booterror.js (hunk 1 above)
Comment (jasoncalabrese) [inline lib/booterror.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2453712997/issues/source?bid=2452379&fileBranchId=3249620#l11)

File: lib/booterror.js (hunk 2 above)
Comment (jasoncalabrese) [inline lib/booterror.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2453712997/issues/source?bid=2452379&fileBranchId=3249620#l8)

**Comment by MilosKozak** [comment]
:+1: this will save lot of our time on support. THANKS

**Pull request merged**